trailblazer/server/api.py

```diff
@@ -72,13 +72,13 @@ def analyses():
 
 @blueprint.route("/analyses/<int:analysis_id>", methods=["GET", "PUT"])
 def analysis(analysis_id):
-    """Display a single analysis."""
+    """Retrieve or update an analysis."""
     analysis: Analysis = store.get_analysis_with_id(analysis_id)
     if analysis is None:
         return abort(404)
 
     if request.method == "PUT":
-        analysis.update(request.json)
+        store.update_analysis(analysis_id=analysis_id, data=request.json)
 
     data = analysis.to_dict()
```

Comment:
slightly unfortunate with the naming clash here. Might be worth to change the former to "body" or something but feel free to disregard.